File: src/main/java/no/nav/pensjon/innsyn/domain/Omsorg.java
package no.nav.pensjon.innsyn.domain;

public class Omsorg {

    private final int ar;
    private final String kilde;
    private final String type;
    private final String status;

    public Omsorg(int ar, String kilde, String type, String status) {
        this.ar = ar;
        this.kilde = kilde;
        this.type = type;
        this.status = status;
    }

    public int getAr() {
        return ar;
    }

    public String getKilde() {
        return kilde;
    }

    public String getType() {
        return type;
    }

    public String getStatus() {
        return status;
    }
}

File: src/main/kotlin/no/nav/pensjon/innsyn/domain/Omsorg.kt
package no.nav.pensjon.innsyn.domain

import javax.persistence.*

@Entity
@Table(name = "T_OMSORG")
@SecondaryTables(
        SecondaryTable(name = "T_K_KILDE_T", pkJoinColumns = [PrimaryKeyJoinColumn(name = "K_KILDE_T")]),
        SecondaryTable(name = "T_K_OMSORG_T", pkJoinColumns = [javax.persistence.PrimaryKeyJoinColumn(name = "K_OMSORG_T")]),
        SecondaryTable(name = "T_K_OMSORG_S", pkJoinColumns = [javax.persistence.PrimaryKeyJoinColumn(name = "K_OMSORG_S")])
)
data class Omsorg(
        @Column(name = "AR")
        val ar: Int,
        @Column(name = "DEKODE", table = "T_K_KILDE_T")
        val kilde: String,
        @Column(name = "DEKODE", table = "T_K_OMSORG_T")
        val type: String,
        @Column(name = "DEKODE", table = "T_K_OMSORG_S")
        val status: String
) {
    @Id
    @Column(name = "PERSON_ID")
    var personId: Int? = null
}